### ISSUE TITLE
Logging module not working after import pynvim

### ISSUE THREAD

**Issue by chemzqm** [neovim/pynvim#373]
``` py
import pynvim
import logging
logfile = '/tmp/coc-jedi.log'
level = logging.DEBUG
logging.basicConfig(filename=logfile, level=level)
logging.debug('abc')
```
Nothing would write to `/tmp/coc-jedi.log`, without `import pynvim`, it works as expected.

**Comment by chemzqm** [comment]
Comment out logging change in `__init__.py` could fix this problem:

```
# Required for python 2.6
#class NullHandler(logging.Handler):
#    def emit(self, record):
#        pass
#
#
#if not logging.root.handlers:
#    logging.root.addHandler(NullHandler())
```

**Comment by justinmk** [comment]
we could skip that section for python3+

**Comment by bfredl** [comment]
Should it just be moved to `setup_logging()`? The pynvim host should set up logging, but pynvim as a lib should leave that to the app using the library.

**Comment by Drllap** [comment]
Is there any workaround for this?

**Comment by nsbradford** [comment]
I've found 2 workarounds for this.

First, remove the NullHandler after you've imported `pynvim` but before configuring your logger:

```python
from pynvim import NullHandler

# Remove the NullHandler from the root logger if it's there.
# see: https://github.com/neovim/pynvim/issues/373
for handler in logging.root.handlers:
    if isinstance(handler, NullHandler):
        logging.root.removeHandler(handler)

# now add your config
logging.basicConfig(level=args.loglevel)
```

Alternatively, you can setup logging before the import of pynvim:
```python
import logging

# Set up your logging.
logging.basicConfig(level=logging.DEBUG)

# Now, import pynvim
import pynvim
```

**Comment by wookayin** [comment]
I checked this issue and can confirm that this has fixed in pynvim 0.5.0-devel (unreleased yet). The workarounds above are valid for pynvim 0.4.3 (and would still be needed for 0.4.3).

FYI, it was fixed in ac03f5c (#492) by dropping python2 support and removing the problematic NullHandler lines.
